### PR TITLE
Add mltrace assert exceptions & logging trigger results

### PR DESCRIPTION
**Pull request by aditim1359**
Within MLTrace, this request adds an MLTrace error exception and adds an mltrace version of the python unittest assert error checking for tests. 

This request also implements the logging for the trigger test results by adding the results to the component run and does the db mirgration.

### CODE REVIEW

File: examples/new_framework_test.py [hunk 1]
```diff
@@ -18,7 +19,9 @@
 
 _identifier = "".join(random.choice(string.ascii_lowercase) for i in range(10))
 
-c = Component("aditi", "test", "test_description")
+# c = Component("aditi", "test", "test_description")
```
Comment (shreyashankar) [inline examples/new_framework_test.py:22]: Delete line

File: mltrace/client.py [hunk 2]
```diff
@@ -529,6 +529,7 @@ def get_component_run_information(component_run_id: str) -> ComponentRun:
     """Returns a ComponentRun object."""
     store = Store(_db_uri)
     cr = store.get_component_run(component_run_id)
+    print("test results from cr: ", cr.test_results)
```
Comment (shreyashankar) [inline mltrace/client.py:532]: remove print stmt

File: mltrace/db/store.py [hunk 3]
```diff
@@ -339,6 +339,7 @@ def commit_component_run(
             f"Committing ComponentRun of type "
             + f'"{component_run.component_name}" to the database.'
         )
+        print("before store: ", component_run.test_results)
```
Comment (shreyashankar) [inline mltrace/db/store.py:342]: remove print stmt

File: mltrace/entities/base_component.py [hunk 4]
```diff
@@ -333,7 +335,16 @@ def wrapper(*args, **kwargs):
                         else inv_user_kwargs[k]: v
                         for k, v in local_vars.items()
                     }
-                    self.afterRun(**after_run_args)
+                    status.update(self.afterRun(**after_run_args))
+
+                # update the component's testStatus, convert status to a json
+                print("status ", status)
```
Comment (shreyashankar) [inline mltrace/entities/base_component.py:341]: remove print

File: mltrace/entities/tests.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+"""
+This file specifies different kinds of tests and the
+functions to be run within each test.
```
Comment (shreyashankar) [inline mltrace/entities/tests.py:3]: Move this file to examples/

File: mltrace/entities/components.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+"""
+This file defines different kinds of components and the
```
Comment (shreyashankar) [inline mltrace/entities/components.py:2]: Move this file to examples/

**Review by shreyashankar — CHANGES_REQUESTED** [review]
Small nits

File: mltrace/entities/components.py (hunk 6 above)
Comment (aditim1359) [reply mltrace/entities/components.py]: done

File: examples/new_framework_test.py (hunk 1 above)
Comment (aditim1359) [reply examples/new_framework_test.py]: done

File: mltrace/client.py (hunk 2 above)
Comment (aditim1359) [reply mltrace/client.py]: done

File: mltrace/db/store.py (hunk 3 above)
Comment (aditim1359) [reply mltrace/db/store.py]: done

File: mltrace/entities/base_component.py (hunk 4 above)
Comment (aditim1359) [reply mltrace/entities/base_component.py]: done

File: mltrace/db/migrations/versions/b86f884647fe_add_test_result_column_to_component_runs.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+Create Date: 2021-11-21 11:59:43.100036
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = 'b86f884647fe'
+down_revision = '331edc89f24b'
```
Comment (shreyashankar) [inline mltrace/db/migrations/versions/b86f884647fe_add_test_result_column_to_component_runs.py:14]: change this down_revision to "8824d15c21d6" (just tested on my computer)

File: mltrace/db/migrations/versions/b86f884647fe_add_test_result_column_to_component_runs.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
+"""add test_result column to component_runs
+
+Revision ID: b86f884647fe
+Revises: 331edc89f24b
```
Comment (shreyashankar) [inline mltrace/db/migrations/versions/b86f884647fe_add_test_result_column_to_component_runs.py:4]: change the down revision to "8824d15c21d6"

**Review by shreyashankar — APPROVED** [review]
Merge after changing the revision numbers

**Pull request merged**